mypyc/ir/ops.py

```diff
@@ -755,6 +759,85 @@ def accept(self, visitor: 'OpVisitor[T]') -> T:
         return visitor.visit_primitive_op(self)
 
 
+@trait
+class IRBuilderInterface:
```

Comment:
I'd prefer to avoid a circular dependency. One plausible-sounding option would be to have an abstract "empty" type here that we'd cast inside primitive op definitions (such as `int_ops.py`) to the concrete IR builder type using some helper function. I.e. here the type would only act as a tag and wouldn't contain any methods. The benefit would be that we can avoid circular dependencies and we don't need to declare all methods here. The drawbacks would be needing to use the cast helper in each callback, and losing a tiny bit of static type safety (which doesn't really matter, I think).

If most primitive op definitions are just simple calls etc., we can have special casing for them, which would allow us to avoid the extra boilerplate.